src/ExpressiveInstaller/Resources/src/ExpressiveSymfonyContainer.php

```diff
@@ -0,0 +1,226 @@
+<?php
+
+namespace App;
+
+use Symfony\Component\DependencyInjection\Container;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Dumper\PhpDumper;
+use Symfony\Component\DependencyInjection\Reference;
+
+/**
+ * Configuration for the Symfony Dependency Injection container.
+ *
+ * This class provides functionality for the following service types:
+ *
+ * - Aliases
+ * - Delegators
+ * - Factories
+ * - Invokable classes
+ * - Services (known instances)
+ */
+class ExpressiveSymfonyContainer
+{
+    const ENABLE_CACHE = 'cache_symfony_container';
+
+    /**
+     * @var array
+     */
+    private $config;
+
+    /**
+     * @var string|null
+     */
+    private $cachedContainerFile;
+
+    /**
+     * @param array $config
+     */
+    public function __construct(
+        array $config,
+        $cachedContainerFile = null
+    ) {
+        $this->config = $config;
+        $this->cachedContainerFile = $cachedContainerFile;
+    }
+
+    /**
+     * return Container
+     */
+    public function create()
+    {
+        $loadFromCache = null !== $this->cachedContainerFile &&
+            file_exists($this->cachedContainerFile) &&
+            isset($this->config[static::ENABLE_CACHE]) &&
+            $this->config[static::ENABLE_CACHE];
+
+        if ($loadFromCache) {
+            return $this->loadFromCache($this->cachedContainerFile);
+        }
+
+        $container = $this->build();
+
+        $this->cacheContainer($container);
+
+        return $container;
```

Comment:
you should not return the container builder. You should also return the optimized container you just generated in the cache